File: app/src/main/java/com/kelsos/mbrc/services/LibraryService.java
package com.kelsos.mbrc.services;

import com.kelsos.mbrc.data.Page;
import com.kelsos.mbrc.data.library.Album;
import com.kelsos.mbrc.data.library.Artist;
import com.kelsos.mbrc.data.library.Genre;
import com.kelsos.mbrc.data.library.Track;
import rx.Observable;

public interface LibraryService {
  Observable<Page<Genre>> getGenres(int offset, int limit);

  Observable<Page<Artist>> getArtists(int offset, int limit);

  Observable<Page<Album>> getAlbums(int offset, int limit);

  Observable<Page<Track>> getTracks(int offset, int limit);
}

File: app/src/main/kotlin/com/kelsos/mbrc/services/LibraryService.kt
package com.kelsos.mbrc.services

import com.kelsos.mbrc.data.Page
import com.kelsos.mbrc.data.library.Album
import com.kelsos.mbrc.data.library.Artist
import com.kelsos.mbrc.data.library.Genre
import com.kelsos.mbrc.data.library.Track
import rx.Observable

interface LibraryService {
  fun getGenres(offset: Int, limit: Int): Observable<Page<Genre>>

  fun getArtists(offset: Int, limit: Int): Observable<Page<Artist>>

  fun getAlbums(offset: Int, limit: Int): Observable<Page<Album>>

  fun getTracks(offset: Int, limit: Int): Observable<Page<Track>>
}
